deploy/eck-operator/values.yaml

```diff
@@ -159,7 +159,12 @@ config:
   exposedNodeLabels: [ "topology.kubernetes.io/.*", "failure-domain.beta.kubernetes.io/.*" ]
 
   # setDefaultSecurityContext determines whether a default security context is set on application containers created by the operator.
-  setDefaultSecurityContext: true
+  # *note* that the default option now is "", which is using "auto-detect" to attempt to set this properly automatically when both running
+  # in an openshift cluster, and a standard kubernetes cluster.  Valid values are as follows:
+  #   ""    : auto detect
```

Comment:
Should we support an additional value called `auto-detect` or `autodetect` that we could use here so we avoid the awkwardness of the empty string value.

Reply:
So, have helm values of ["on","off","auto-detect"], which translates to ["on","off",""] in the flags of the operator?  I assume this also means that if its unset, that it's inherently "auto-detect"?  I think that makes good sense.  Will do.  Let me know if my assumptions here are incorrect.

Reply:
I was thinking more of `true`, `false`, `auto-detect` and no translation between Helm and operator parameters. The operator parameters should also be these three if that makes sense. Or is there a benefit of having the "" parameter that I am missing?

Reply:
I've updated this to only allow those 3x options.   `setDefaultSecurityContext` being unset, or "" will fail with a syntax error at `strconv.ParseBool(setDefaultSecurityContext)`